kubernetes/services/rabbitmq.yml.j2

```diff
@@ -0,0 +1,70 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: rabbitmq
+spec:
+  replicas:             1
+  revisionHistoryLimit: 1
+  template:
+    metadata:
+      labels:
+        run: rabbitmq
+    spec:
+      hostname: rabbitmq-1
+      containers:
+        - image:           rabbitmq:3.7.4
+          name:            rabbitmq
+          imagePullPolicy: Always
+          env:
+            - name:  RABBITMQ_MNESIA_BASE
+              value: /srv/storage/mnesia
+          resources:
+            requests:
+              memory: "1200Mi"
+              cpu:    "600m"
+            limits:
+              memory: "1600Mi"
+              cpu:    "900m"
+          volumeMounts:
+            - mountPath:  /srv/storage/mnesia/
+              name:       queue-storage
+          livenessProbe:
+            exec:
+              command:
+              - rabbitmqctl
+              - status
+            initialDelaySeconds: 120
+            timeoutSeconds: 5
+            failureThreshold: 6
+          readinessProbe:
+            exec:
+              command:
+              - rabbitmqctl
+              - status
+            initialDelaySeconds: 10
+            timeoutSeconds: 3
+            periodSeconds: 5
+      volumes:
+        - name: queue-storage
+          gcePersistentDisk:
+            pdName: {{ queue_disk }}
+            fsType: ext4
+      # securityContext allows you to add permission for pod and mounted volumes to specific group.
+      # We need this permission to create files and directories on mounted persistent disk.
+      # Rabbitmq user has reserved 999 ID
+      # We use this user when creating files and directories.
+      securityContext:
+        fsGroup: 999
```

Comment:
1) > Rabbitmq user has reserved 999 ID

    `fsGroup` is the ID of the group, not of the user

2) Are you sure the ID is 999?

    Here's how they create the user and the group: https://github.com/docker-library/rabbitmq/blob/master/3.7-rc/alpine/Dockerfile#L4
    ``` Dockerfile
    # add our user and group first to make sure their IDs get assigned consistently, regardless of whatever dependencies get added
    RUN addgroup -S rabbitmq && adduser -S -h /var/lib/rabbitmq -G rabbitmq rabbitmq
    ```

    They do not have it hard-coded to 999 and their instructions will not result in that ID. You can verify with:

    ``` bash
    docker run \
        --interactive \
        --tty \
        --rm \
        alpine:3.7 \
            sh -c \
                "addgroup -S rabbitmq && \
                adduser -S -h /var/lib/rabbitmq -G rabbitmq rabbitmq && \
                cat /etc/passwd | grep rabbitmq && \
                cat /etc/group | grep rabbitmq"
    ``` 

    The output is:
    ```
    rabbitmq:x:100:101:Linux User,,,:/var/lib/rabbitmq:/bin/false
    rabbitmq:x:101:rabbitmq
    ```    
    So their user has ID `100` and group has ID `101`.

Reply:
The output from rabbitmq pod:
`/etc/passwd`:
```
rabbitmq:x:999:999::/var/lib/rabbitmq:
```
`/etc/group`:
```
rabbitmq:x:999:

```

Reply:
But how? Did I look at the wrong Docker image?

Reply:
Yes, you are looking at a wrong image, we use this: https://github.com/docker-library/rabbitmq/blob/master/3.7/debian/Dockerfile

Reply:
OK, so:
1) Why not use the alpine image?
2) Definitely add a comment that says that this ID depends on the base image. That's not obvious.

Reply:
1. I don't remember exactly, but I think the problem was related with feature and stability:
- https://docs.docker.com/samples/library/rabbitmq/#enabling-hipe
- https://docs.docker.com/samples/library/rabbitmq/#rabbitmqalpine
2. Ok, I added a comment.

Reply:
In that case let's switch to Alpine. The warning is only relevant if we would want to add some software that heavily depends on GNU libc to the image. We're not adding anything so for us alpine is fine (at least for now).